node/src/gcp/mod.rs

```diff
@@ -0,0 +1,398 @@
+        let kind: String = format!("{}-{}", T::kind(), self.env);
+        let req = RunQueryRequest {
+            database_id: Some("".to_string()),
+            partition_id: Default::default(),
+            read_options: Default::default(),
+            query: Some(Query {
+                projection: None,
+                kind: Some(vec![KindExpression { name: Some(kind) }]),
+                filter,
+                order: None,
+                distinct_on: Some(vec![]),
+                start_cursor: None,
+                end_cursor: None,
+                offset: None,
+                limit: None,
+            }),
+            gql_query: None,
+        };
+        let (_hyper_resp, query_resp) = self
+            .datastore
+            .projects()
+            .run_query(req, &self.project_id)
+            .doit()
+            .await?;
+        let batch = query_resp.batch.ok_or_else(|| {
+            DatastoreStorageError::FetchEntitiesError(
+                "Could not retrieve batch while fetching entities".to_string(),
+            )
+        })?;
+
+        batch.entity_results.ok_or_else(|| {
+            DatastoreStorageError::FetchEntitiesError(
+                "Could not retrieve entity results while fetching entities".to_string(),
+            )
+        })
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn delete<T: IntoValue + KeyKind>(&self, value: T) -> DatastoreResult<()> {
+        let kind: String = format!("{}-{}", T::kind(), self.env);
+        let mut entity = Entity::from_value(value.into_value())?;
+        let path_element = entity
+            .key
+            .as_mut()
+            .and_then(|k| k.path.as_mut())
+            .and_then(|p| p.first_mut());
+        if let Some(path_element) = path_element {
+            // We can't create multiple datastore databases in GCP, so we have to suffix
+            // type kinds with env (`dev`, `prod`).
+            path_element.kind = Some(kind)
+        }
+
+        let request = CommitRequest {
+            database_id: Some("".to_string()),
+            mode: Some(String::from("NON_TRANSACTIONAL")),
+            mutations: Some(vec![Mutation {
+                insert: None,
+                delete: entity.key,
+                update: None,
+                base_version: None,
+                upsert: None,
+                update_time: None,
+            }]),
+            single_use_transaction: None,
+            transaction: None,
+        };
+        tracing::debug!(?request);
+        let (_, response) = self
+            .datastore
+            .projects()
+            .commit(request, &self.project_id)
+            .doit()
+            .await?;
+        tracing::debug!(?response, "received response");
+
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct GcpService {
+    pub project_id: String,
+    pub datastore: Option<DatastoreService>,
+    pub secret_manager: Option<SecretManagerService>,
```

Comment:
why are these options? Would be unwieldy to use later when we go and directly reference them everytime?